content/blog/2019/root-transition/index.md

```diff
@@ -0,0 +1,142 @@
+you are recommended to follow the procedure listed below to upgrade.
+
+If you are currently using the mTLS feature in Istio with self-signed certificates,
+please follow the procedure and check whether you will be affected.
+
+## Procedure
+
+### Step 0. Check when your root certificate will expire
+
+Download this [script](TODO) on a machine that has kubectl access to the cluster.
+
+{{< text bash>}}
+$ ./root-transition.sh check
+...
+===YOU HAVE 30 DAYS BEFORE THE ROOT CERT EXPIRED!=====
+{{< /text >}}
+
+Please run the following steps to upgrade immediately to avoid system outages, before your root certificate expires.
+
+### Step 1. Do the root certificate transition
+
+Run the following command to do the root certificate transition.
+
+{{< text bash>}}
+$ ./root_transition.sh transition
+create new ca cert, with trust domain as  cluster.abc
+Wed Jun  5 19:11:15 PDT 2019 delete old ca secret
+secret "istio-ca-secret" deleted
+Wed Jun  5 19:11:15 PDT 2019 create new ca secret
+secret/istio-ca-secret created
+pod "istio-citadel-8574b88bcd-j7v2d" deleted
+Wed Jun  5 19:11:18 PDT 2019 restarted Citadel, checking status
+NAME                             READY     STATUS    RESTARTS   AGE
+istio-citadel-8574b88bcd-l2g74   1/1       Running   0          3s
+New root certificate:
+Certificate:
+    Data:
+        ...
+        Validity
+            Not Before: Jun  6 03:24:43 2019 GMT
+            Not After : Jun  3 03:24:43 2029 GMT
+        Subject: O = " cluster.abc"
+        ...
+{{< /text >}}
+
+After the transition, the Envoy sidecars may be hot-restarted. This may have some impact on your traffic.
+Please refer to 
+[here](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/hot_restart#arch-overview-hot-restart) and
+[here](https://blog.envoyproxy.io/envoy-hot-restart-1d16b14555b5) for more details.
+
+### Step 2. Verify the new workload certificates are generated
+
+Run the following command to verify that all the new certificates are generated. If not, wait a minute and run it again.
+
+{{< text bash>}}
+$ ./root_transition.sh verify
+This script checks the current root CA certificate is propagated to all the Istio-managed workload secrets in the cluster.                    
+Root cert MD5 is 8fa8229ab89122edba73706e49a55e4c
+Checking namespace: default
+  Secret default.istio.default is updated.
+  Secret default.istio.sleep is updated.
+Checking namespace: istio-system
+  Secret istio-system.istio.default is updated.
+  ...
+------All Istio keys and certificates are updated in secret!
+{{< /text >}}
+
+### Step 3. Install Istio 1.1.8
+
+Note that this step is necessary, to make sure the control plane components and sidecar Envoys all reload the new certificates and keys.
+
+Upgrade your control plane and Envoy sidecars to 1.1.8. You can refer to this
+[doc](https://istio.io/docs/setup/kubernetes/upgrade/steps/)
+to do the upgrades.
+
+### Step 4. Verify the new workload certificates are loaded by Envoy
```

Comment:
`1. Verify the new workload certificates are loaded by Envoy:`

Reply:
If we use the automatic numbering approach, does it start from 1 instead of step "0" as we are using here?

Reply:
yes it starts from 1.